### PR TITLE
Fix incremental materialization doesnt drop temp table

### LINKED ISSUE #195: Incremental materialization doesn't drop temp table
### Expected behavior

When something goes wrong during incremental materialization, rerunning dbt should fix the problem.

### Actual behavior

When something goes wrong during incremental materialization, the temp table isn't dropped so once something goes wrong, dbt can never recover and needs someone to manually drop the temp table for things to work again.

### Steps To Reproduce

1. Run incremental model that fails for operational reasons (e.g. Trino being unstable). 
2. Run it again.

Dbt-trino creates temporary relation to hold the data that it then appends. That gets cleaned up on success but not before run, so if things go wrong the relation stays there, until cleaned up manually.

### Log output/Screenshots

```
10:03:25  Database Error in model <redacted> (models/<redacted>.sql)
10:03:25    TrinoUserError(type=USER_ERROR, name=TABLE_ALREADY_EXISTS, message="line 1:1: Destination table '<redacted>' already exists", query_id=20221211_095653_13629_vchpy)

```

### Operating System

MacOS

### dbt version

1.3.0

### Trino Server version

393-e.1

### Python version

3.8.10

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by damian3031**
## Overview
resolves #195
Fix incremental materialization doesn't drop temp table.
Change views_enabled to true by default.
Add test for get_incremental_tmp_relation_type macro.

### CODE REVIEW

**Comment by wrb2** [comment]
Hi,

Thank you for this!

Would it also make sense to switch `views_enabled` to true by default, as @mdesmet said in the issue?

File: dbt/include/trino/macros/materializations/incremental.sql [hunk 1]
```diff
@@ -9,7 +9,7 @@
        we can save the model query definition as a view instead,
        for faster overall incremental processing.
   #} */
-  {%- set views_enabled = config.get('views_enabled', false) -%}
+  {%- set views_enabled = config.get('views_enabled', true) -%}
```
Comment (hovaesco) [inline dbt/include/trino/macros/materializations/incremental.sql:12]: I would move it to a separate commit.

Comment (damian3031) [reply dbt/include/trino/macros/materializations/incremental.sql:12]: moved

**Review by mdesmet — APPROVED** [review]
LGTM, do you see any way to test the `views_enabled` flag?

**Comment by damian3031** [comment]
> LGTM, do you see any way to test the `views_enabled` flag?

@mdesmet maybe we could test [get_incremental_tmp_relation_type](https://github.com/starburstdata/dbt-trino/blob/fix-incr-materialization-doesnt-drop-temp-table/dbt/include/trino/macros/materializations/incremental.sql#L1) macro for expected output?

**Comment by mdesmet** [comment]
I think you can merge the commits on setting views_enabled to true. I don't think it needs a separate changie entry.

**Comment by mdesmet** [comment]
@damian3031 : please execute `pre-commit install` to ensure you execute the checks and formatting on commit.

**Review by mdesmet — APPROVED** [review]

**Pull request merged**